src/libAtomVM/defaultatoms.c

```diff
@@ -23,10 +23,15 @@
 #include <stdlib.h>
 #include <string.h>
 
+// About X macro: https://en.wikipedia.org/wiki/X_macro
+#define X(name, lenstr, str) \
+    _Static_assert(lenstr[0] == sizeof(str) - 1, "Macro length mismatch for " # name);
+#include "defaultatoms.def"
+#undef X
+
 void defaultatoms_init(GlobalContext *glb)
 {
 
-// About X macro: https://en.wikipedia.org/wiki/X_macro
```

Comment:
I personally found this breadcrumb very helpful to understand how these macros work. I suggest we keep it for future contributors who are unfamiliar with this technique.